core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1317,7 +1326,9 @@ class ReplicaManager(val config: KafkaConfig,
             partitionOpt.foreach { partition =>
               val currentLeaderEpoch = partition.getLeaderEpoch
               val requestLeaderEpoch = partitionState.leaderEpoch
-              if (requestLeaderEpoch > currentLeaderEpoch) {
+              val currentZkVersion = partition.getZkVersion
+              val requestZkVersion = partitionState.zkVersion
+              if (requestLeaderEpoch > currentLeaderEpoch || requestZkVersion > currentZkVersion) {
```

Comment:
Do we still need this change? I think we are trying to keep the current approach where the controller bumps the leader epoch for any controller-initiated change.

Reply:
No, we don't need this anymore. This was added so a LeaderAndIsr could update the Partition state without a leader epoch bump, but we don't have that flow anymore so we can revert this.